examples/widgets/slider_demo.py

```diff
@@ -5,63 +5,70 @@
 
 Using the slider widget to control visual properties of your plot.
 
-In this example, a slider is used to choose the frequency of a sine
-wave. You can control many continuously-varying properties of your plot in
-this way.
+In this example, sliders are used to control the frequency and amplitude of
+a sine wave. You can control many continuously-varying properties of your plot
+in this way.
+
+For a more detailed example of value snapping see
+:doc:`/gallery/widgets/slider_snap_demo`.
+
+For an example of using a `matplotlib.widgets.RangeSlider` to define a range
+of values see :doc:`/gallery/widgets/range_slider`.
 """
 import numpy as np
 import matplotlib.pyplot as plt
-from matplotlib.widgets import Slider, Button, RadioButtons
+from matplotlib.widgets import Slider, Button
+
+
+def fxn(t, amp, freq):
+    return amp * np.sin(2 * np.pi * freq * t)
 
-fig, ax = plt.subplots()
-plt.subplots_adjust(left=0.25, bottom=0.25)
 t = np.arange(0.0, 1.0, 0.001)
+
+# Define initial parameters
 a0 = 5
 f0 = 3
-delta_f = 5.0
-s = a0 * np.sin(2 * np.pi * f0 * t)
-l, = plt.plot(t, s, lw=2)
-ax.margins(x=0)
+
+# Create the figure and the `~.Line2D` that we will manipulate
+fig, ax = plt.subplots()
+line, = plt.plot(t, fxn(t, a0, f0), lw=2)
 
 axcolor = 'lightgoldenrodyellow'
+ax.margins(x=0)
+
+# adjust the main plot to make room for the sliders
+plt.subplots_adjust(left=0.25, bottom=0.25)
+
+# Make a horizontal slider to control the frequency.
+# This slider will snap to discrete values as defind by ``valstep``.
 axfreq = plt.axes([0.25, 0.1, 0.65, 0.03], facecolor=axcolor)
-axamp = plt.axes([0.25, 0.15, 0.65, 0.03], facecolor=axcolor)
+freq_slider = Slider(axfreq, 'Freq', 0.1, 30.0, valinit=f0, valstep=5.0)
 
-sfreq = Slider(axfreq, 'Freq', 0.1, 30.0, valinit=f0, valstep=delta_f)
-samp = Slider(axamp, 'Amp', 0.1, 10.0, valinit=a0)
+# Make a vertically oriented slider to control the amplitude
+axamp = plt.axes([0.1, 0.15, 0.03, 0.65], facecolor=axcolor)
```

Comment:
Though I find two horizontal sliders visually more appealing, I see the point of showing a vertical slider.

I'd align the vertical slider with the plot Axes and scale it's width so that it's rendered the same width as the horizontal slider in a figure with default size and aspect.

```suggestion
axamp = plt.axes([0.1, 0.25, 0.0225, 0.63], facecolor=axcolor)
```

I know that these are magic numbers, but here we are with the widgets. There have been magic numbers before. The only difference is that these have been chosen more carefully for a better visual appearance.

Reply:
This does look better I will use.

Reply:
I think ideally it would be easy to specify the size of sliders in units like inches because that's what I've almost always ended up doing when I used them. Except that I do that in a very roundabout way.

Something like
`Slider(fig, xy, width, height, ...)`

where at least width and height are in `inches`, less obvious what I would want `xy` to be though.